android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -199,6 +220,116 @@ private boolean isAllowedUrl(Uri loadingUrl) {
+            case 501 -> "Not Implemented";
+            case 502 -> "Bad Gateway";
+            case 503 -> "Service Unavailable";
+            case 504 -> "Gateway Timeout";
+            case 505 -> "HTTP Version Not Supported";
+            default -> "Unknown";
+        };
+    }
+
+    private WebResourceResponse handleCapacitorHttpRequest(WebResourceRequest request) throws IOException {
+        boolean isHttps =
+            request.getUrl().getPath() != null && request.getUrl().getPath().startsWith(Bridge.CAPACITOR_HTTPS_INTERCEPTOR_START);
+
+        String urlString = request
+            .getUrl()
+            .toString()
+            .replace(bridge.getLocalUrl(), isHttps ? "https:/" : "http:/")
+            .replace(Bridge.CAPACITOR_HTTP_INTERCEPTOR_START, "")
+            .replace(Bridge.CAPACITOR_HTTPS_INTERCEPTOR_START, "");
+        URL url = new URL(urlString);
+        JSObject headers = new JSObject();
+
+        for (Map.Entry<String, String> header : request.getRequestHeaders().entrySet()) {
+            headers.put(header.getKey(), header.getValue());
+        }
+
+        HttpRequestHandler.HttpURLConnectionBuilder connectionBuilder = new HttpRequestHandler.HttpURLConnectionBuilder()
+            .setUrl(url)
+            .setMethod(request.getMethod())
+            .setHeaders(headers)
+            .openConnection();
+
+        CapacitorHttpUrlConnection connection = connectionBuilder.build();
+
+        if (!isDomainExcludedFromSSL(bridge, url)) {
+            connection.setSSLSocketFactory(bridge);
+        }
+
+        connection.connect();
+
+        String mimeType = null;
+        String encoding = null;
+        Map<String, String> responseHeaders = new LinkedHashMap<>();
+        for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
+            StringBuilder builder = new StringBuilder();
+            for (String value : entry.getValue()) {
+                builder.append(value);
+                builder.append(", ");
+            }
+            builder.setLength(builder.length() - 2);
+
+            if ("Content-Type".equalsIgnoreCase(entry.getKey())) {
+                String[] contentTypeParts = builder.toString().split(";");
+                mimeType = contentTypeParts[0].trim();
+                if (contentTypeParts.length > 1) {
+                    String[] encodingParts = contentTypeParts[1].split("=");
+                    if (encodingParts.length > 1) {
+                        encoding = encodingParts[1].trim();
+                    }
+                }
+            } else {
+                responseHeaders.put(entry.getKey(), builder.toString());
+            }
+        }
+
+        // Using Live Reload
+        if (bridge.getServerUrl() != null) {
+            responseHeaders.put("Access-Control-Allow-Origin", bridge.getServerUrl());
+            responseHeaders.put("Access-Control-Allow-Methods", "GET, HEAD, OPTIONS, TRACE");
+        }
```

Comment:
```suggestion
```
I've just realized we don't really need CORS headers on Android since the server url is the same than the proxy url